### ISSUE TITLE
Datenschutzerklärung nicht editierbar

### ISSUE THREAD

**Issue by MaxGitHubAccount** [foodcoopshop/foodcoopshop#676]
Pro: Es gibt eine Datenschutzerklärung und sie wird automatisch durch die eingestellten Daten befüllt
Cons: Man kann die generierte Erklärung nicht überschreiben

Warum sollte man das tun?
Bspw. Der Datenschutzbeauftragte ist nicht immer der aus Österreich.

Workaround bis der Bug gefixt ist, selbst eine Seite anlegen und in apache über eine Rewrite Rule weiterleiten

> RewriteEngine On
> RewriteRule ^/datenschutzerklaerung$ https://%{SERVER_NAME}/Pfad-zur-Datenschutzerklärung [NC]

**Comment by MaxGitHubAccount** [comment]
Ich finde es aber seltsam, dass häufig Issues die nur augenscheinlich gelöst sind, geschlossen werden. Insbesondere, da du selbst schreibst, dass hier etwas zu tun wäre. Offen heißt ja nicht, dass sie eine hohe Prio haben, werden aber direkt allen angezeigt, die sich fragen was vllt noch zu tun wären. Am Ende ist es aber natürlich dein Projekt und dein Stil das Projekt zu führen.

Die Datenschutzbehörde in Österreich ist nicht das einzige Problem. Man kann also im Zweifel die ganze Erklärung in die custom_config.php packen. Eine Datenschutzerklärung kann sehr individuell sein und es kann sich bspw. auch anbieten auf eine zentrale Seite außerhalb des Shops zu verlinken, wenn man bspw. mehrere Seiten unter der gleichen Domain führt und dann mehrere Daten anfallen.
Außerdem denke ich ist es hilfreich, wenn der Shop die Datenschutzerklärung nicht unveränderbar vorgibt. Ich finde es gut, dass dadurch für alle ersichtlich ist, was der Shop erhebt - jedoch ändert sich das Rechtsverständnis und was in so eine Erklärung rein sollte mit der Zeit und obwohl es eine Verordnung ist auch je Land.

**Comment by mrothauer** [comment]
> Ich finde es aber seltsam, dass häufig Issues die nur augenscheinlich gelöst sind, geschlossen werden. Insbesondere, da du selbst schreibst, dass hier etwas zu tun wäre. Offen heißt ja nicht, dass sie eine hohe Prio haben, werden aber direkt allen angezeigt, die sich fragen was vllt noch zu tun wären. Am Ende ist es aber natürlich dein Projekt und dein Stil das Projekt zu führen.

Ja, da gebe ich dir recht. Natürlich ist das Issue nicht gelöst, nur weil ich es geschlossen habe. Ich versuche nur, den Überblick zu behalten und bei vielen offenen Issues - auch mit niedriger Prio - ist das schwierig.

Dass die Datenschutzerklärung nicht editierbar ist (außer mit der von mir beschriebenen Methode) hat schon seinen Sinn - die Foodcoops müssen sich nicht drum kümmern und haben einen Gesetzestext, der im Wesentlichen rechtskonform ist.

Wenn du mir einen fertigen Text für Deutschland schickst (bzw. einen PR stellst), kann ich gerne eine Land-Einstellung einbauen und die Datenschutzerklärung für Deutschland aufnehmen.

**Comment by MaxGitHubAccount** [comment]
Im Grunde ist die Auskunft wenn man nur einen Shop betreibt natürlich bis auf den Abschnitt mit Österreich korrekt. Ich habe für unsere Erklärung folgendes Tool vom LfDI Baden-Württemberg genutzt: https://www.baden-wuerttemberg.datenschutz.de/ds-gvo.clever/
Leider kann man die Österreichische Aufsicht nicht einfach durch eine Deutsche austauschen - ist ländersache ¯\_(ツ)_/¯
D.h. was man natürlich machen könnte wäre alles lassen wie es ist + das er aus dem Standort erkennt wer zuständig ist oder bspw. per Drop Down Österreich oder Bundesländer. In dem Generator ist ja nicht mal die Adresse der Aufsicht benannt, wäre also auch nicht zu kompliziert.

**Comment by mrothauer** [comment]
Um deinen Use-Case am einfachsten abzudecken, würde ich eine neue Einstellung "Land / Region" als Dropdown mit zwei Einträgen einbinden: Österreich und Baden-Württemberg. Kann ja dann wachsen und mit mehr Regionen befüllt werden. Bitte sag mir noch genau, welche Stelle in der aktuellen Datenschutzerklärung weg muss und durch welchen Text ersetzt gehört, dann setze ich das um. Wird dann Teil von v3.3 (erscheint im Herbst 2021).
Inhalt der Datenschutzerklärung als Textarea ist komplizierter umzusetzen, weil der Text dann in die Datenbank wandern müsste - und das möchte ich aus diversen Gründen (hat mit den Übersetzungen zu tun) vermeiden.

**Comment by MaxGitHubAccount** [comment]
Keinen Stress - wir haben bei uns nun eine andere Lösung umgesetzt mit einer Datenschutzerklärung für alle Dienste die wir anbieten. Die einfachste Lösung wenn du nur Texte wechselst wäre vermutlich, wenn du das wie du schreibst per Drop Down machst und dann immer nur die Verantwortliche Stelle änderst. Ich kann dir die Verantwortlichen Stellen für ganz Deutschland liefern (bzw. einfach den Generator nehmen und einmal jedes Land auswählen ^^)

**Comment by mrothauer** [comment]
> Edit: Eine andere Lösung wäre sonst vielleicht noch, wenn man eine im Shop erstelle Page auswählen könnte.

Ich glaube, das ist die beste Idee. Ich werde es im Laufe des Sommers umsetzen, damit es in v3.3 dann mit dabei ist.

Magst du mir noch sagen, von welcher Foodcoop du bist? Mich interessiert natürlich, wo die Software überall verwendet wird. Auch gerne nicht öffentlich per Mail: office@foodcoopshop.com
